src/incubator/TextField/types.ts

```diff
@@ -203,6 +207,10 @@ export type TextFieldProps = MarginModifiers &
      * Should validate when losing focus of TextField
      */
     validateOnBlur?: boolean;
+    /**
+     * Callback for when field validated and failed
+     */
+    onValidateFailed?: (failedValidatorIndex: number) => void;
```

Comment:
please also add it to the `src/incubator/TextField/textField.api.json` file, for the docs :)